### ISSUE TITLE
Problem about Filters in multi-search endpoint with federation

### ISSUE THREAD

**Issue by honhimW** [meilisearch/meilisearch#4857]
**Describe the bug**
Searching result not as expected.

**To Reproduce**
- Create an index with the movies.json data.
- Update the index settings about filterableAttributes: 'id', 'title', 'overview', 'genres'
- Update the index settings about sortableAttributes: 'id'
- Search with params
```http
POST /multi-search

{
  "queries": [
    {
      "q": "'The Matrix'",
      "attributesToHighlight": [
        "*"
      ],
      "facets": [],
      "highlightPreTag": "<pre>",
      "highlightPostTag": "</pre>",
      "filter": [
        "id < 100"
      ],
      "sort": [
        "id:desc"
      ],
      "showRankingScore": true,
      "attributesToSearchOn": [
        "title", "overview"
      ],
      "indexUid": "{movies}"
    }
  ],
  "federation": {
    "limit": 20,
    "offset": 0
  }
}
```

**Expected behavior**
A result set in reverse order of id, and id less than 100.
Obviously `estimatedTotalHits` should less than 100.

**Meilisearch version:**
_v1.10.0-rc.1_

**Additional context**
When changing the filter value `id < 1000`, the result changes(Obviously the filter is working).

When attempting multiple page flips (offset: offset += 20, limit: 20), duplicate documents appear.

**Comment by dureuill** [comment]
Related to #4717: reverting its commits fixes the issue.

The link to #4848 is that similar optimizations to #4717 were implemented in #4713 and #4682 (which is in v1.9)

**Comment by curquiza** [comment]
Fixed by #4858 will be released in rc3
